FridaTests/MacOsTargetTests.swift

```diff
@@ -0,0 +1,162 @@
+        return url.deletingLastPathComponent()
+    }
+    
+    private func getSourceRoot() -> URL {
+        let fileManager = FileManager.default
+        var url = URL(fileURLWithPath: #file)
+        while fileManager.fileExists(atPath: url.appendingPathComponent(".git").path) == false {
+            url = url.deletingLastPathComponent()
+            
+            if url.pathComponents.count == 0 {
+                url = URL(fileURLWithPath: #file).deletingLastPathComponent().deletingLastPathComponent()
+                break
+            }
+        }
+        
+        return url
+    }
+    
+    private func spawnSessionWith(script scriptContents: String) {
+        let localDeviceExpectation = expectation(description: "Found local device.")
+        var localDeviceMaybe: Device?
+        manager.enumerateDevices { result in
+            let devices = try! result()
+            let localDevice = devices.filter { $0.kind == Device.Kind.local }.first!
+            localDeviceMaybe = localDevice
+            localDeviceExpectation.fulfill()
+        }
+        waitForExpectations(timeout: 2, handler: nil)
+        XCTAssertNotNil(localDeviceMaybe)
+        guard localDeviceMaybe != nil else { return }
+        localDevice = localDeviceMaybe
+        
+        
+        let binary = getProductsConfigurationDirectory().appendingPathComponent("TestTarget")
+        let spawnExpectation = expectation(description: "Spawned binary.")
+        var pidMaybe: UInt?
+        localDevice.spawn(binary.path) { result in
+            do {
+                pidMaybe = try result()
+            } catch let error {
+                print("Couldn't launch, error: \(error)")
+            }
+            
+            spawnExpectation.fulfill()
+        }
+        waitForExpectations(timeout: 2, handler: nil)
+        XCTAssertNotNil(pidMaybe)
+        guard pidMaybe != nil else { return }
+        pid = pidMaybe
+        
+        let attachExpectation = expectation(description: "Attached to binary.")
+        var sessionMaybe: Session?
+        localDevice.attach(pid) { (result) in
+            sessionMaybe = try? result()
+            attachExpectation.fulfill()
+        }
+        waitForExpectations(timeout: 2, handler: nil)
+        XCTAssertNotNil(sessionMaybe)
+        guard sessionMaybe != nil else { return }
+        session = sessionMaybe
+        
+        let createScriptExpectation = expectation(description: "Created script.")
+        var scriptMaybe: Script?
+        session.createScript(scriptContents) { (result) in
+            scriptMaybe = try! result()
+            createScriptExpectation.fulfill()
+        }
+        waitForExpectations(timeout: 2, handler: nil)
+        XCTAssertNotNil(scriptMaybe)
+        guard scriptMaybe != nil else { return }
+        script = scriptMaybe
+        script.delegate = scriptDelegate
+    }
+    
+    override func tearDown() {
+        manager.close()
+    }
+    
+    func testReceiveMessage() {
```

Comment:
Feels like the two tests should go to the top of the class, right after the member variables and `teardown`. (Newspaper ordering.)